src/mlpack/methods/ann/layer/convolution_impl.hpp

```diff
@@ -118,10 +133,10 @@ void Convolution<
   inputTemp = arma::cube(const_cast<arma::Mat<eT>&&>(input).memptr(),
       inputWidth, inputHeight, inSize * batchSize, false, false);
 
-  if (padW != 0 || padH != 0)
+  if (padWLeft != 0 || padWRight != 0 || padHTop != 0 || padHBottom != 0)
   {
-    inputPaddedTemp.set_size(inputTemp.n_rows + padW * 2,
-        inputTemp.n_cols + padH * 2, inputTemp.n_slices);
+    inputPaddedTemp.set_size(inputTemp.n_rows + padWLeft + padWRight,
```

Comment:
Would be great if we could add a test as well.

Reply:
Actually I am doing this for `Inception Layer`, so I think we will know whether it is wrong or not. Logically It seems correct. Let me know what you think.

Reply:
Sorry for the slow response, ideally we can write an independent test for the adjusted conv operation, similar to https://github.com/mlpack/mlpack/blob/444559d71e1d29ef518c836aa571b40850a8755c/src/mlpack/tests/convolution_test.cpp#L124.